singer_sdk/authenticators.py

```diff
@@ -95,6 +95,18 @@ def auth_params(self) -> dict:
         """
         return self._auth_params or {}
 
+    def authenticate_request(self, request: requests.Request) -> None:
+        """Authenticate a request.
+
+        Args:
+            request: A `request object`_.
+
+        .. _request object:
+            https://requests.readthedocs.io/en/latest/api/#requests.Request
+        """
+        request.headers.update(self.auth_headers or {})
+        request.params.update(self.auth_params or {})
```

Comment:
The properties already default to `{}`, so there's no need to do it here too. The properties themselves also seem redundant - why not just use the attributes `auth_headers` and `auth_params` directly, and change them into properties in the future if they need special handling?

```suggestion
        request.headers.update(self.auth_headers)
        request.params.update(self.auth_params)
```

Reply:
> why not just use the attributes `auth_headers` and `auth_params` directly, and change them into properties in the future if they need special handling?

@WillDaSilva I agree. `_auth_headers` should just be `auth_headers` in some of the constructors. Do you think it's worth trying that in this PR?

Reply:
@edgarrmondragon Let's focus on getting this PR merged in today in time for the release. That change is just cleanup that can wait and/or be done in a separate PR.